sympy/combinatorics/perm_groups.py

```diff
@@ -694,6 +694,44 @@ def basic_transversals(self):
             self.schreier_sims()
         return self._transversals
 
+    def composition_series(self):
+        """
+        Add docs
+
+        """
+        if not self.is_solvable:
+            raise NotImplementedError('Group should be solvable')
+        der = self.derived_series()
+        series = [self]
+
+        for i in range(len(der)-1):
+            comp = []
+            low = der[i+1]
+            gns = low.generators
+            while low != der[i]:
+                while True:
+                    elm = der[i].random()
```

Comment:
I think that it would suffice to take the generators of der[i] instead of random elements.